polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -201,13 +201,22 @@ impl Artifacts {
 	/// Create an empty table and populate it with valid artifacts as [`ArtifactState::Prepared`],
 	/// if any. The existing caches will be checked by their file name to determine whether they are
 	/// valid, e.g., matching the current node version. The ones deemed invalid will be pruned.
+	///
+	/// Create the cache directory on-disk if it doesn't exist.
 	pub async fn new_and_prune(cache_path: &Path) -> Self {
 		let mut artifacts = Self { inner: HashMap::new() };
-		artifacts.insert_and_prune(cache_path).await;
+		match artifacts.insert_and_prune(cache_path).await {
+			Ok(()) => (),
+			Err(err) => gum::error!(
+				target: LOG_TARGET,
+				"could not initialize artifacts cache: {}",
+				err,
+			),
+		}
```

Comment:
nit: No need to explicitly match here

```suggestion
		artifacts.insert_and_prune(cache_path).await.map_err(|err| gum::error!(
				target: LOG_TARGET,
				"could not initialize artifacts cache: {err}",
			));
```